File: dev/backend/schemas/project.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional


class ProjectBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=100, description="プロジェクト名")
    description: Optional[str] = Field(
        None, max_length=1000, description="プロジェクト説明"
    )


class ProjectCreate(ProjectBase):
    pass


class ProjectUpdate(BaseModel):
    name: Optional[str] = Field(
        None, min_length=1, max_length=100, description="プロジェクト名"
    )
    description: Optional[str] = Field(
        None, max_length=1000, description="プロジェクト説明"
    )


class ProjectResponse(ProjectBase):
    uuid: str = Field(..., description="プロジェクトUUID")
    user_uuid: str = Field(..., description="ユーザーUUID")
    is_deleted: bool = Field(..., description="削除フラグ")
    created_at: datetime = Field(..., description="作成日時")
    updated_at: datetime = Field(..., description="更新日時")

    class Config:
        from_attributes = True


class ProjectListResponse(BaseModel):
    projects: list[ProjectResponse] = Field(..., description="プロジェクトリスト")
    total: int = Field(..., description="総件数")


class ProjectDeleteResponse(BaseModel):
    message: str = Field(..., description="削除完了メッセージ")
    uuid: str = Field(..., description="削除されたプロジェクトUUID")

File: dev/frontend/src/features/project/types/project.ts
export interface Project {
  uuid: string;
  user_uuid: string;
  name: string;
  description?: string;
  is_deleted: boolean;
  created_at: string;
  updated_at: string;
}

export interface ProjectCreate {
  name: string;
  description?: string;
}

export interface ProjectUpdate {
  name?: string;
  description?: string;
}

export interface ProjectListResponse {
  projects: Project[];
  total: number;
}

export interface ProjectDeleteResponse {
  message: string;
  uuid: string;
}

export interface ProjectState {
  projects: Project[];
  deletedProjects: Project[];
  currentProject: Project | null;
  loading: boolean;
  error: string | null;
  total: number;
}
